services/hanke-service/src/integrationTest/kotlin/fi/hel/haitaton/hanke/application/ApplicationServiceITest.kt

```diff
@@ -597,6 +598,58 @@ class ApplicationServiceITest : DatabaseTest() {
             }
         }
 
+        @Test
+        fun `when application is in Allu should initialize access for new contacts`() {
+            val cableReport =
+                createCableReportApplicationData(
+                    areas = listOf(aleksanterinpatsas),
+                    customerWithContacts = hakijaCustomerContact,
+                    contractorWithContacts = suorittajaCustomerContact,
+                )
+            val application =
+                hankeService.generateHankeWithApplication(
+                    CableReportWithoutHanke(CABLE_REPORT, cableReport),
+                    USERNAME,
+                )
+            mockSendAllu(applicationRepository.findById(application.id!!).orElseThrow())
+            val capturedNotifications = mutableListOf<ApplicationNotificationData>()
+            with(cableReportServiceAllu) {
+                justRun { update(21, any()) }
+                every { getApplicationInformation(any()) } returns createAlluApplicationResponse(21)
+                justRun { addAttachment(21, any()) }
+            }
+            with(emailSenderService) {
+                justRun { sendHankeInvitationEmail(any()) }
+                justRun { sendApplicationNotificationEmail(capture(capturedNotifications)) }
+            }
+
+            applicationService.updateApplicationData(
+                application.id!!,
+                cableReport.copy(
+                    representativeWithContacts = asianHoitajaCustomerContact,
+                    propertyDeveloperWithContacts = rakennuttajaCustomerContact,
+                    contractorWithContacts =
+                        suorittajaCustomerContact.changeContactEmails("new.mail@foo.fi")
+                ),
+                USERNAME
+            )
+
+            assertThat(capturedNotifications).hasSize(3)
+            assertThat(capturedNotifications)
+                .areValid(application.applicationType, application.hankeTunnus)
+            verifySequence {
+                with(cableReportServiceAllu) {
+                    getApplicationInformation(any())
+                    update(21, any())
+                    addAttachment(any(), any())
+                }
```

Comment:
Is it so bad to repeat `cableReportServiceAllu` three times? Using control structures increases cognitive load, i.e. makes code harder to understand.

Reply:
Removed withs.